File: Modul-4/Soal-4/PRAK404-2410817320001-NazlaSalsabila.py
def kalkulator():
    nama = "NAZLASALSABILA"
    while True:
        print("Pilih program:\n1. Penjumlahan\n2. Pengurangan\n3. Perkalian\n4. Pembagian\n5. Exit")
        try:
            pilihan = int(input("Masukkan Pilihan: "))
            if pilihan == 5:
                print(f"Terimakasih, telah menggunakan kalkulator {nama}")
                break
            if pilihan not in [1, 2, 3, 4]:
                print("Input anda salah, silahkan coba lagi")
                continue
            a, b = float(input("Masukkan nilai pertama: ")), float(input("Masukkan nilai kedua: "))
            hasil = a + b if pilihan == 1 else a - b if pilihan == 2 else a * b if pilihan == 3 else (a / b if b != 0 else "Pembagian dengan nol tidak diperbolehkan!")
            print(f"Hasil {'penjumlahan' if pilihan == 1 else 'pengurangan' if pilihan == 2 else 'perkalian' if pilihan == 3 else 'pembagian'} antara {a:.2f} dengan {b:.2f} adalah {hasil:.2f}" if isinstance(hasil, float) else hasil)
        except ValueError:
            print("Input anda salah, silahkan coba lagi")

kalkulator()

File: Modul-4/Soal-4/PRAK404-2410817320001-NazlaSalsabila.c
#include <stdio.h>

int main() {
    int pilihan;
    float a, b, hasil;
    while (1) {
        printf("Pilih program:\n1. Penjumlahan\n2. Pengurangan\n3. Perkalian\n4. Pembagian\n5. Exit\nMasukkan Pilihan: ");
        scanf("%d", &pilihan);
        if (pilihan == 5) {
            printf("Terimakasih, telah menggunakan kalkulator NAZLASALSABILA\n");
            break;
        } else if (pilihan < 1 || pilihan > 5) {
            printf("Input anda salah, silahkan coba lagi\n");
            continue;
        }
        printf("Masukkan nilai pertama: ");
        scanf("%f", &a);
        printf("Masukkan nilai kedua: ");
        scanf("%f", &b);
        if (pilihan == 1) hasil = a + b, printf("Hasil penjumlahan %.2f dengan %.2f adalah %.2f\n", a, b, hasil);
        else if (pilihan == 2) hasil = a - b, printf("Hasil pengurangan %.2f dengan %.2f adalah %.2f\n", a, b, hasil);
        else if (pilihan == 3) hasil = a * b, printf("Hasil perkalian %.2f dengan %.2f adalah %.2f\n", a, b, hasil);
        else if (pilihan == 4) b == 0 ? printf("Pembagian dengan nol tidak diperbolehkan!\n") : (hasil = a / b, printf("Hasil pembagian %.2f dengan %.2f adalah %.2f\n", a, b, hasil));
    }
    return 0;
}